homeassistant/components/light/__init__.py

```diff
@@ -369,3 +369,15 @@ def state_attributes(self):
     def supported_features(self):
         """Flag supported features."""
         return 0
+
+    @asyncio.coroutine
+    def async_added_to_hass(self, state):
+        """Component added, restore_state using platforms."""
+        if hasattr(self, 'async_restore_state'):
+            params = {key: state.attr[key] for key in (
+                ATTR_BRIGHTNESS, ATTR_RGB_COLOR, ATTR_XY_COLOR,
+                ATTR_COLOR_TEMP, ATTR_COLOR_NAME, ATTR_WHITE_VALUE
+            ) if key in state.attr}
+            params['is_on'] = state.state == STATE_ON
+            # pylint: disable=no-member
+            return (yield from self.async_restore_state(params))
```

Comment:
Each has to handle our own.

`async_added_to_hass` is a generic hook. It has 0 knowledge that it is being used of restoration. That is how we keep the system decoupled.

We also no longer have to keep track of restoration, it's now handled by platforms so platforms are fully aware of the data. The extra attribute was only needed before, because platforms were not aware that they could exist in the frontend.

With the current approach we will no longer have any states being restored without the actual device being found.